### PR TITLE
Add Splink Blog to the docs

### LINKED ISSUE #1215: [FEAT] Add Changelog
### Is your proposal related to a problem?

With new features being added regularly, we need a better way of communicating the latest developments with users.

### Describe the solution you'd like

A new section in the docs to highlight the latest features being added to splink. This could take the form of a formal changelog simply linking to the relevant parts of the docs or a more rounded blog where we can highlight the top stories and then link to parts of the docs with more detail.

### PR DESCRIPTION
**Pull request by RossKen**
### Type of PR

- [ ] BUG
- [ ] FEAT
- [ ] MAINT
- [X] DOC

### Is your Pull Request linked to an existing Issue or Pull Request?

Closed #1215 

### Give a brief description for the solution you have provided

Add blog to the documentation site. Primary usage will be to highlight latest features to users, but may end up being more general in future.

### PR Checklist

- [ ] Added documentation for changes
- [ ] Added feature to example notebooks at tutorial in [splink_demos](https://github.com/moj-analytical-services/splink_demos) (if appropriate)
- [ ] Added tests (if appropriate)
- [ ] Made changes based off the latest version of Splink
- [ ] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

File: docs/blog/posts/2023-05-24-latest_developments.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+<!-- more -->
+
+Latest Splink version: [v3.9.4](https://github.com/moj-analytical-services/splink/releases/tag/v3.9.4)
+
+
+## :rocket: Massive speed gains in EM training
+
+There’s now an option to make EM training much faster - in one example we’ve seen at 1000x fold speedup.  Kudos to external contributor [@aymonwuolanne](https://github.com/moj-analytical-services/splink/pull/1369) from the Australian Bureau of Statistics! 
+
+To make use of this, set the `estimate_without_term_frequencies` parameter to True, for example:
```
Comment (ADBond) [inline docs/blog/posts/2023-05-24-latest_developments.md:27]: Maybe ',' -> ';' here? Initially read as though setting the parameter to True is one example of how to do this

File: docs/blog/posts/2023-05-24-latest_developments.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+```py
+linker.estimate_parameters_using_expectation_maximisation(..., estimate_without_term_frequencies=True)
```
Comment (ADBond) [inline docs/blog/posts/2023-05-24-latest_developments.md:30]: Do we have anywhere to link with caveat about possible parameter accuracy? Just that if reading this I might wonder about why I would have to opt-in to get such a big speedup

File: docs/blog/posts/2023-05-24-latest_developments.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+brl.exact_match_rule("date_of_birth")
+```
+
+Check out these new functions in the [BRL Documentation](https://moj-analytical-services.github.io/splink/blocking_rule_library.html) as well as some new [Blocking Topic Guides](https://moj-analytical-services.github.io/splink/topic_guides/blocking/blocking_rules.html) to better explain what Blocking Rules are, how they are used in Splink, and how to choose them.
+
+Keep a look out, as there is more improvements in the pipeline for Blocking in the coming months!
+
+## :elephant: Postgres Support
+
+With a massive thanks to external contributor [@hanselmm](https://github.com/moj-analytical-services/splink/pull/1191), Splink now supports :simple-postgresql: Postgres. To get started, check out the [Postgres Topic Guide](https://moj-analytical-services.github.io/splink/topic_guides/backends/postgres.html).
```
Comment (ADBond) [inline docs/blog/posts/2023-05-24-latest_developments.md:64]: `hanselmm` -> `hanslemm`

**Review by ADBond — COMMENTED** [review]
This looks great! Really like the style and think it will be a great addition.

As far as content goes, I'm not sure how far back you had in mind for this, but one thing I thought might be nice to include is the regex match/null functionality - it's quite a cool feature that can otherwise quite easily slip by unnoticed, as it is just a couple of extra function params, and this feels like a useful place to highlight

**Comment by RossKen** [comment]
> This looks great! Really like the style and think it will be a great addition.
> 
> As far as content goes, I'm not sure how far back you had in mind for this, but one thing I thought might be nice to include is the regex match/null functionality - it's quite a cool feature that can otherwise quite easily slip by unnoticed, as it is just a couple of extra function params, and this feels like a useful place to highlight

Great idea! I have made a note to add this

File: docs/blog/posts/2023-05-24-latest_developments.md (hunk 1 above)
Comment (RossKen) [reply docs/blog/posts/2023-05-24-latest_developments.md:27]: Changed

File: docs/blog/posts/2023-05-24-latest_developments.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+```py
+import splink.duckdb.blocking_rule_library as brl
+
+brl.exact_match_rule("date_of_birth")
+```
+
+Check out these new functions in the [BRL Documentation](https://moj-analytical-services.github.io/splink/blocking_rule_library.html) as well as some new [Blocking Topic Guides](https://moj-analytical-services.github.io/splink/topic_guides/blocking/blocking_rules.html) to better explain what Blocking Rules are, how they are used in Splink, and how to choose them.
+
+Keep a look out, as there is more improvements in the pipeline for Blocking in the coming months!
```
[the remark this answers is not in the document]
Comment (RossKen) [reply docs/blog/posts/2023-05-24-latest_developments.md:60]: fixed

File: docs/blog/posts/2023-05-24-latest_developments.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+## :elephant: Postgres Support
+
+With a massive thanks to external contributor [@hanselmm](https://github.com/moj-analytical-services/splink/pull/1191), Splink now supports :simple-postgresql: Postgres. To get started, check out the [Postgres Topic Guide](https://moj-analytical-services.github.io/splink/topic_guides/backends/postgres.html).
+
+## :label: Clerical Labelling Tool (beta)
+
+Clerical labelling is an important tool for [generating performance metrics](https://moj-analytical-services.github.io/splink/demos/tutorials/07_Quality_assurance.html) for linkage models (False Positive Rate, Recall, Precision etc.).
+
+Splink now has a (beta) GUI for clerical labelling which produces labels in a form that can be easily ingested into Splink to generate these performance metrics. Check out the [example tool](https://robinlinacre.com/splink_example_charts/example_charts/splink3/labelling_tool_dedupe_only.html), linked [Pull Request](https://github.com/moj-analytical-services/splink/pull/1208), and some previous tweets:
```
[the remark this answers is not in the document]
Comment (RossKen) [reply docs/blog/posts/2023-05-24-latest_developments.md:70]: We don't unfortunately - the [PR](https://github.com/moj-analytical-services/splink/pull/1208/files#diff-a5ee695236b5c718ce2980a43a114eb41971b3d5f030449c8b016c9040d671be) didn't have anything on the docs side. Perhaps I should write up a brief doc just to outline what is going on that we can point to here.

File: docs/blog/posts/2023-05-24-latest_developments.md (hunk 2 above)
Comment (RossKen) [reply docs/blog/posts/2023-05-24-latest_developments.md:30]: Good point - do you think we should just include a caveat here? Or add a caveat in a more permanent home in the docs and link to it from here?

File: docs/blog/posts/2023-05-24-latest_developments.md (hunk 3 above)
Comment (RossKen) [reply docs/blog/posts/2023-05-24-latest_developments.md:64]: fixed

File: docs/blog/posts/2023-05-24-latest_developments.md (hunk 2 above)
Comment (RossKen) [reply docs/blog/posts/2023-05-24-latest_developments.md:30]: It is mentioned in the docs of the [parameter in estimate_parameters_using_expectation_maximisation](https://moj-analytical-services.github.io/splink/linkerest.html#splink.linker.Linker.estimate_parameters_using_expectation_maximisation) so my first instinct is to add a caveat here and refer to these API docs.

File: docs/blog/posts/2023-05-24-latest_developments.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (RossKen) [reply docs/blog/posts/2023-05-24-latest_developments.md:70]: Have opened up an issue to add this #1471

File: docs/blog/posts/2023-05-24-latest_developments.md (hunk 2 above)
Comment (RossKen) [reply docs/blog/posts/2023-05-24-latest_developments.md:30]: Yep, totally makes sense

**Comment by RossKen** [comment]
@ADBond I have added the EM caveat as well as the regex stuff and a few other small tweaks. Would you mind having a quick check over?

On the clerical labelling stuff - I am slightly torn. I think we should have some documentation, as you have rightly pointed out. However, I would like to get this published relatively soon as I think it will be valuable for people, and the docs will take time to write up/get reviewed properly. I'm also running out of steam on the docs front and think I need to go off and do some actual coding for a bit over the next week or so. Is it possibly worth leaving the labelling tool for the next blog when we have properly documented it? I know we probably want to get feedback on it sooner rather than later though...

@RobinL would also be keen to hear any thoughts you have on this?

**Comment by RobinL** [comment]
> @ADBond I have added the EM caveat as well as the regex stuff and a few other small tweaks. Would you mind having a quick check over?
> 
> On the clerical labelling stuff - I am slightly torn. I think we should have some documentation, as you have rightly pointed out. However, I would like to get this published relatively soon as I think it will be valuable for people, and the docs will take time to write up/get reviewed properly. I'm also running out of steam on the docs front and think I need to go off and do some actual coding for a bit over the next week or so. Is it possibly worth leaving the labelling tool for the next blog when we have properly documented it? I know we probably want to get feedback on it sooner rather than later though...
> 
> @RobinL would also be keen to hear any thoughts you have on this?

Yeah I'd probably err on the side of getting this out rather than delaying it until we write more docs, especially since there's the example that people can take a look at without actually writing any code.  It's not ideal, but if people really want to use this immediately, they can always ask a question on the discussion forum

**Comment by RossKen** [comment]
> > @ADBond I have added the EM caveat as well as the regex stuff and a few other small tweaks. Would you mind having a quick check over?
> > On the clerical labelling stuff - I am slightly torn. I think we should have some documentation, as you have rightly pointed out. However, I would like to get this published relatively soon as I think it will be valuable for people, and the docs will take time to write up/get reviewed properly. I'm also running out of steam on the docs front and think I need to go off and do some actual coding for a bit over the next week or so. Is it possibly worth leaving the labelling tool for the next blog when we have properly documented it? I know we probably want to get feedback on it sooner rather than later though...
> > @RobinL would also be keen to hear any thoughts you have on this?
> 
> Yeah I'd probably err on the side of getting this out rather than delaying it until we write more docs, especially since there's the example that people can take a look at without actually writing any code. It's not ideal, but if people really want to use this immediately, they can always ask a question on the discussion forum

Ok, thanks both. I will edit to make it more obvious what state of WIP this is in. @RobinL are you happy for me to include a link to the gist you posted on the docs issue?

**Comment by RobinL** [comment]
Yeah, totally fine to link to the gist.  Not sure, but might also be worth linking to the [pr](https://github.com/moj-analytical-services/splink/pull/1208) since it contains quite a lot of information about the background to the tool

**Review by ADBond — APPROVED** [review]
Great, thanks for those changes. Looks good - happy to go ahead without the labelling docs

**Comment by RossKen** [comment]
Great, thanks both! I will make a minor wording edit then we are good to go.

**Pull request merged**